linked_list/linked_list.py

```diff
@@ -10,70 +10,173 @@ def __init__(self, value, next_node = None):
 class LinkedList:
     def __init__(self):
       self.head = None # keep the head private. Not accessible outside this class
+      self.size = 0
+      self.tail = None
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(n)
     def get_first(self):
-        pass
+        if self.head == None:
+            return None
+        return self.head.value    
 
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(self, value):
-        pass
+        current = self.head
+        if self.head == None:
+            self.head = Node(value)
+            self.tail = self.head
+            self.size += 1
+        else:
+            new_node = Node(value)
+            new_node.next = current
+            current.previous = new_node
+            self.head = new_node
+            self.size += 1

```

Comment:
Could be more compact

```suggestion
        self.head = Node(value, self.head)
        self.size += 1
        if self.tail is None:
            self.tail = self.head
```